ADAL/src/ADAuthenticationContext+Internal.m

```diff
@@ -181,8 +181,9 @@ + (BOOL)canHandleResponse:(NSURL *)response
         sourceApplication:(NSString *)sourceApplication
 {
 #if TARGET_OS_IPHONE
-    BOOL isResponseFromBroker = [self isResponseFromBroker:sourceApplication response:response];
-    if (!isResponseFromBroker) { return NO; }
+    // sourceApplication could be nil, we want to return early if we know for sure response is not from broker
```

Comment:
should we also verify that response starts with Redirect URI? That way we can stop early for unknown responses.

Reply:
addressed :)